### PR TITLE
Extract Kotlin snippets

### PR DESCRIPTION
**Pull request by ok300**
Extracted the Kotlin snippets into a simple Kotlin MPP project, and referenced them from the docs.

### CODE REVIEW

**Comment by ademar111190** [comment]
Shouldn't it be on sdk examples repo? https://github.com/breez/breez-sdk-examples

**Review by dangeross — COMMENTED** [review]
Looking good, just some comments around naming

**Comment by ok300** [comment]
> Shouldn't it be on sdk examples repo? https://github.com/breez/breez-sdk-examples

@ademar111190 The code snippets are referenced from the markdown docs, so they have to be in this repo. The snippets must also be aligned across languages (e.g. show how to do one thing across languages), and we don't have fully fledged apps in the example repo for all languages yet. Long-term it might make sense to merge the examples repo and this one, but for now IMO it's ok to keep them separate.

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/LnurlAuth.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    fun lnurl_auth(sdk: BlockingBreezServices) {
+        // ANCHOR: lnurl-auth
+        // Endpoint can also be of the form:
+        // keyauth://domain.com/auth?key=val
+        val lnurlAuthUrl = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7..."
+        try {
+            val inputType = parseInput(lnurlAuthUrl)
+            if (inputType is InputType.LnUrlAuth) {
+                when (val result = sdk.lnurlAuth(inputType.data)) {
+                    LnUrlCallbackStatus.Ok -> {} // Log.v("Breez", "Successfully authenticated")
```
[the remark this answers is not in the document]
Comment (ok300) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/LnurlAuth.kt]: Done

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ReceiveOnchain.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        } catch (e: Exception) {
+            // handle error
+        }
+        // ANCHOR_END: list-refundables
+    }
+
+    fun execute_refund(sdk: BlockingBreezServices) {
+        // ANCHOR: execute-refund
+        val swapAddress = "..."
+        val destinationAddress = "..."
```
[the remark this answers is not in the document]
Comment (ok300) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ReceiveOnchain.kt]: Done

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/FiatCurrencies.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            val fiatRatesMap = sdk.fetchFiatRates()
+        } catch (e: Exception) {
+            // handle error
+        }
+        // ANCHOR_END: fetch-fiat-rates
+    }
+
+    fun get_fiat_currencies_and_rates(sdk: BlockingBreezServices) {
+        // ANCHOR: get-fiat-currencies-and-rates
+        fun fiatCurrenciesAndRate(): Map<FiatCurrency, Rate> = try {
```
[the remark this answers is not in the document]
Comment (ok300) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/FiatCurrencies.kt]: True, function in a function was a bit confusing. I was trying to get the same snippet as https://sdk-doc.breez.technology/guide/fiat_currencies.html (Kotlin tab, last section).

Fixed in https://github.com/breez/breez-sdk-docs/pull/101#issuecomment-1816286650

**Comment by JssDWt** [comment]
> The main reason is that it's a very niche way of using our fiat methods. It essentially combines and sorts the outputs of `sdk.list_fiat_currencies()` and `sdk.fetch_fiat_rates()`. IMO its probably overkill for an API documentation, especially since already document how to use each of the 2 API calls.

Nice. I was thinking the same. 
If we would need logic like that it could make more sense to expose another api endpoint to get the data in that form.

**Pull request merged**